src/plots/cartesian/axis_defaults.js

```diff
@@ -161,6 +163,27 @@ function rangebreaksDefaults(itemIn, itemOut, containerOut) {
                 itemOut.bounds = itemOut.bounds.slice(0, 2);
             }
 
+            var dfltPattern = '';
+            var i, q;
+            for(i = 0; i < bnds.length; i++) {
+                q = indexOfDay(bnds[i]);
+                if(q !== -1) {
+                    dfltPattern = DAY_OF_WEEK;
+                    break;
+                }
+            }
+            var pattern = coerce('pattern', dfltPattern);
+
+            if(pattern === DAY_OF_WEEK) {
+                for(i = 0; i < bnds.length; i++) {
+                    q = indexOfDay(bnds[i]);
+                    if(q !== -1) {
+                        // convert to integers i.e 'Monday' --> 1
+                        itemOut.bounds[i] = bnds[i] = q;
+                    }
```

Comment:
Do we also want to validate other values here? ie turn `'0'` through `'6'` into numbers and reject all other strings, numbers outside that range, and non-integers?

Reply:
cc: @nicolaskruchten

Reply:
We can ignore this for now, but at some point it may be nice to play with what happens when users enter bad data, and whether validating it here would help them figure out how to fix it.

Reply:
OK. I think I could fix that in this PR.

Reply:
Done in https://github.com/plotly/plotly.js/pull/4661/commits/e481a13deba3322d6279dd4f0acddc04f297ebbd.